### PR TITLE
[MM-20603] Migrate components/team_settings to TypeScript

### PR DESCRIPTION
**Pull request by djanda97**
#### Summary

In this PR:
* Migrated: `components/team_settings/index.js --> index.ts`
* Migrated: `components/team_settings/team_settings.jsx --> team_settings.tsx`
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/13471

### CODE REVIEW

**Comment by djanda97** [comment]
I get the following error in my IDE when hovering over `GeneralTab` in `team_settings.tsx`:

`Type '{ team: Record<string, unknown>; activeSection: string; updateSection: () => void; closeModal: () => void; collapseModal: () => void; }' is missing the following properties from type 'InferPropsInner<Pick<{ updateSection: Validator<(...args: any[]) => any>; team: Validator<object>; activeSection: Validator<string>; closeModal: Validator<(...args: any[]) => any>; collapseModal: Validator<...>; maxFileSize: Validator<...>; actions: Validator<...>; canInviteTeamMembers: Validator<...>; }, RequiredKe...': maxFileSize, canInviteTeamMembers, actions`

My guess for why this error is appearing is that `components/team_general_tab` requires more props than are being supplied to `GeneralTab` in `team_settings`.

Any advice for how to remedy this?

File: components/team_settings/team_settings.tsx [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
+};
+
+const TeamSettings = ({
+    activeTab = '',
+    activeSection = '',
+    updateSection,
+    closeModal,
+    collapseModal,
+    team,
+}: Props): JSX.Element | null | undefined => {
```
Comment (nevyangelova) [inline components/team_settings/team_settings.tsx:25]: Wonder in which cases will this be undefined?

**Comment by djanda97** [comment]
@devinbinnie Since the consumed component is relatively large (800+ LOC), I opted to just remove `isRequired` from a few of the props in `team_general_tab.jsx`.

File: components/team_settings/team_settings.tsx (hunk 1 above)
Comment (djanda97) [reply components/team_settings/team_settings.tsx]: Mistakenly added. Removed in my recent commits.

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @djanda97!

**Review by jgilliam17 — APPROVED** [review]
Thank you @djanda97 
Tested, looks good to merge.
- Verified team settings - as before.

**Pull request merged**